### ISSUE TITLE
llvm: Allow using cmake for dependency

### ISSUE THREAD

**Issue by dcbaker** [mesonbuild/meson#4861]
We know have a cmake dependency finder, and LLVM can be found using cmake. It should be possible to find LLVM using cmake, for mesa there are some caveats for it's use but this should be investigated and turned on.

**Comment by mensinda** [comment]
From what I can tell this should be possible. However, this might require minor adjustments to the CMake dependency backend to use a different `CMakeLists.txt`. This way, all (required) CMake variables can be extracted:

```cmake
find_package(LLVM REQUIRED CONFIG)
set(MESON_SOME_LLVM_VAR ${SOME_LLVM_VAR})
```

In this case the value of `MESON_SOME_LLVM_VAR` can be retrieved with `get_cmake_var('MESON_SOME_LLVM_VAR')`. Accessing `SOME_LLVM_VAR` directly can be a bit dodgy because of the CMake trace limitations. Also, the CMake files seem to define some targets, which can be easily extracted with the existing meson code.

I have never worked with LLVM, so I am not an expert on that issue.

**Comment by dcbaker** [comment]
LLVM is a giant pile of pain as a dependency. Upstream very much expects you to vendor. I really wish we didn't rely on it in mesa :/

**Comment by mensinda** [comment]
@dcbaker I have implemented a [CMake backend](https://github.com/mensinda/meson/tree/cmakeLLVM) for LLVM. Would you mind trying it out? From what I can tell this should work, but this is my first time working on something LLVM related, so I could have missed something.

No pull request yet, because this branch is based on #4914, which isn't merged yet.

**Comment by dcbaker** [comment]
I wont have time until Monday probably, but I will look at it.

**Comment by mensinda** [comment]
Sure thanks. No rush. I will also create a PR once #4914 is merged.

**Comment by dcbaker** [comment]
I think the other thing I'll want to add for mesa is a generic getter for dependencies, but that's another issue.

**Comment by dcbaker** [comment]
I took a peek at it, there were a couple of things I noticed missing that I need for mesa: The optional modules, shared vs static linking, and a way to query some of the command line interfaces like rtti.
